client/state/plugins/premium/reducer.js

```diff
@@ -66,20 +65,26 @@ export const plugins = withSchemaValidation( pluginInstructionSchema, ( state =
 				} );
 			}
 			return state;
-		case SERIALIZE:
-			// Save the error state as a string message.
-			return mapValues( state, ( pluginList ) =>
+		default:
+			return state;
+	}
+};
+
+export const plugins = withSchemaValidation(
+	pluginInstructionSchema,
+	withPersistence( pluginsReducer, {
+		// Save the error state as a string message and omit the `key` field.
+		serialize: ( state ) =>
+			mapValues( state, ( pluginList ) =>
 				pluginList.map( ( item ) => {
-					if ( item.error !== null ) {
-						return Object.assign( {}, item, { error: item.error.toString() } );
+					if ( item.error != null ) {
```

Comment:
I'd suggest being explicit here. I've seen `!= null` cause pain before. I'd very much prefer specifically checking for `null` and `undefined`.

Reply:
I disagree here. `== null` is widely considered a "good part" of the language and even the ESLint `eqeqeq` rule has an exception that allows `==` here and nowhere else.

I changed this particular condition to just `if ( item.error )`. We don't need any special treatment for falsy values like `false` , `0` or `''` here.

Reply:
I'll disagree with you too 😉  There's a reason why we recommend strict checks in our coding standards. For a large codebase that we need to maintain for years, there's nothing better than being explicit in what we'd like to do in a certain line of code. We can never be certain if the `== null` was intentional or not. While if we specifically check for `null` and `undefined`, it'll be more than clear that it's intentional.